versioned/storage/common-tests/src/main/java/org/projectnessie/versioned/storage/commontests/AbstractBasePersistTests.java

```diff
@@ -1233,4 +1261,92 @@ public void nullHandlingInArrays() throws ObjTooLargeException, ObjNotFoundExcep
   public static String randomContentId() {
     return randomUUID().toString();
   }
+
+  @Test
+  public void conditionalDelete() throws Exception {
+    VersionedTestObj v1 =
+        VersionedTestObj.builder().id(randomObjId()).someValue("initial").versionToken("1").build();
+    VersionedTestObj v2 =
+        VersionedTestObj.builder().from(v1).someValue("version 2").versionToken("2").build();
+
+    // non-existing - delete must not succeed
+    soft.assertThat(persist.deleteConditional(v1)).isFalse();
+
+    soft.assertThatThrownBy(() -> persist.fetchObj(v1.id()))
+        .isInstanceOf(ObjNotFoundException.class);
+    soft.assertThat(persist.storeObj(v1)).isTrue();
+    soft.assertThat(persist.fetchObj(v1.id())).isEqualTo(v1);
+
+    // exists, but different version - delete must not succeed
+    soft.assertThat(persist.deleteConditional(v2)).isFalse();
+    soft.assertThat(persist.fetchObj(v1.id())).isEqualTo(v1);
+    // exists, same version - delete must succeed
+    soft.assertThat(persist.deleteConditional(v1)).isTrue();
+    long t = System.currentTimeMillis();
+
+    soft.assertThatThrownBy(() -> persist.fetchObj(v1.id()))
+        .isInstanceOf(ObjNotFoundException.class);
+
+    if ("Cassandra".equals(persist.name())) {
```

Comment:
Nit: maybe extract to a method, it's repeated twice.

Reply:
done